src/loaders/abstract_loader.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABCMeta, abstractmethod
+from typing import Iterator
+
+from src.models.catalog.video_info import VideoMetaInfo
+from src.models.storage.batch import FrameBatch
+from src.models.storage.frame import Frame
+
+
+class AbstractVideoLoader(metaclass=ABCMeta):
+    """
+    Abstract class for defining video loader. All other video loaders use this
+    abstract class. Video loader are expected fetch the videos from storage
+    and return the frames in an iterative manner.
+
+    Attributes:
+        video_metadata (VideoMetaInfo): Object containing metadata of the video
+        batch_size (int, optional): No. of frames to fetch in batch from video
+        skip_frames (int, optional): Number of frames to be skipped
+                                     while fetching the video
+        offset (int, optional): Start frame location in video
+        limit (int, optional): Number of frames needed from the video
+        shard (int, optional): Shard number to load from if sharded
```

Comment:
shard_id?

Reply:
Delaying it for the next PR. Hopefully will be fixed in the later PR.